### PR TITLE
Reframe `--locked` and `--frozen` as `--check` operations for `uv lock`

### PR DESCRIPTION
**Pull request by zanieb**
Closes https://github.com/astral-sh/uv/issues/7639

### CODE REVIEW

**Comment by johnthagen** [comment]
@zanieb As a Poetry user, I think `uv lock --check` is a lot easier to understand as I was definitely confused by the meaning of `uv lock --locked` when trying to find the `uv` equivalent for `poetry check --lock`. Big 👍 from me.

**Comment by samypr100** [comment]
> @zanieb As a Poetry user, I think `uv lock --check` is a lot easier to understand as I was definitely confused by the meaning of `uv lock --locked` when trying to find the `uv` equivalent for `poetry check --lock`. Big 👍 from me.

 @johnthagen this reminds me of https://github.com/python-poetry/poetry/pull/8015 😂

**Comment by zanieb** [comment]
:D yeah. I think we're fairly far off from a `uv check` command but I would _presume_ it does something like this? ref #9653 

`uv check` might be expected to focus on your _code_ (given the amount of static analysis tooling we happen to have) so maybe that'd be a separate command? Multiple commands to check the project are definitely annoying though.

File: docs/reference/cli.md [hunk 1]
```diff
@@ -1790,6 +1790,18 @@ uv lock [OPTIONS]
 <p>To view the location of the cache directory, run <code>uv cache dir</code>.</p>
 
 <p>May also be set with the <code>UV_CACHE_DIR</code> environment variable.</p>
+</dd><dt><code>--check</code></dt><dd><p>Check if the lockfile is up-to-date.</p>
+
+<p>Asserts that the <code>uv.lock</code> would remain unchanged after a resolution. If the lockfile is missing or needs to be updated, uv will exit with an error.</p>
+
+<p>Equivalent to <code>--locked</code>.</p>
+
+<p>May also be set with the <code>UV_LOCKED</code> environment variable.</p>
```
Comment (charliermarsh) [inline docs/reference/cli.md:1799]: The env var / CLI name mismatch here is a bit strange, but oh well.

**Review by charliermarsh — APPROVED** [review]

File: docs/reference/cli.md (hunk 1 above)
Comment (zanieb) [reply docs/reference/cli.md:1799]: Yeah, I noticed that too but hopefully the "Equivalent to" call-out above is helpful. `UV_LOCK_CHECK` doesn't really make sense to set via the environment anyway.

**Pull request merged**